client/modules/Booking/Booking.js

```diff
@@ -235,59 +248,99 @@ export class Booking extends Component {
-                Confirm
-              </Button>
-              </Form>
-      </div>
+          <FormGroup>
+            <Label for="ccv">Please confirm your credit card CVV</Label>
+            <Input
+              type="text"
+              name="ccv"
+              id="ccv"
+              placeholder="CVV"
+              value={this.state.ccv}
+              onBlur={() => this.handleBlur("ccv")}
+              onChange={this.handleCcvChange.bind(this)}
+            />
+            <FormFeedback>A valid CVV is required.</FormFeedback>
+          </FormGroup>
+          <Button outline color="success" className={styles.wideBtn}>
+            Confirm
+          </Button>
+        </Form>
+      </Container>
     );
   }
 
   booked() {
     return (
-      <div className={styles.body}>
+      <Container className={styles.body}>
         <h1 className={styles.title}>Booking success</h1>
-        <p>
-          <Link to="/">Click here to return home</Link>
-          <br />
-        </p>
-      </div>
+        <Row>
+          <Col sm="12" md="6">
+            <hr />
+            <Card>
+              <CardHeader tag="h5">
+                {this.state.selectedCar.year} {this.state.selectedCar.make}{" "}
+                {this.state.selectedCar.model} ({this.state.selectedCar.colour}){" "}
+                <br />
+                <span className="text-muted">
+                  {this.state.selectedLocation.name}
+                </span>
+              </CardHeader>
+              <CardBody>
+                <CardText>
+                  Vehicle type: {this.state.selectedCar.vehicleType.name},{" "}
+                  {this.state.selectedCar.doors} doors<br />
+                  Seats: {this.state.selectedCar.seats}
+                  <br />
+                  Hourly rate: ${this.state.selectedCar.vehicleType.hourlyRate.toFixed(
+                    2
+                  )}
+                  <br />
+                  Registration: {this.state.selectedCar.rego}
+                </CardText>
+              </CardBody>
+            </Card>
+          </Col>
+          <Col sm="12" md="6">
+            <hr />
+            <Card>
+              <CardHeader tag="h5">
+                Booked from <br />
+                <span className="text-muted">{" "+this.state.startDate.format("MMMM Do YYYY HH:mm").toString()+" "}</span>
+                to
+                <span className="text-muted">{" "+this.state.endDate.format("MMMM Do YYYY HH:mm").toString()+" "}</span>
+              </CardHeader>
+              <CardBody>
+                <CardText>Booking Cost : ${this.state.cost}</CardText>
+              </CardBody>
+            </Card>
+          </Col>
+        </Row>
+        <Row>
+          <Col>
+            <Link to="/">Click here to return home</Link>
+          </Col>
+        </Row>
+      </Container>
     );
   }
 
+  //Attempt at callback function
+  //results in read only error
+  returnTime(time)
+  {
+    this.startDate=moment(time);
+  }
```

Comment:
you need to use `setState` to update the state - and not reassign it directly:

```
returnTime(time) {
  this.setState({ startDate: moment(time) })
}
```

Also, I believe the argument you pass in here is already an instance of moment (e.g. wrapped with a `moment` call earlier) - if this is true, then you don't need to double-wrap it again

Reply:
There are two different variables. this.state.startDate wouldn't work. It wouldnt recognise setState